Service/AlgoliaCredentialsManager.php

```diff
@@ -46,22 +46,24 @@ public function checkCredentialsWithSearchOnlyAPIKey(int $storeId = null): bool
      * @param string $class
      * @param int|null $storeId
      * @return void
-     * @throws NoSuchEntityException
      */
     public function displayErrorMessage(string $class, ?int $storeId = null): void
```

Comment:
I've observed we tend to call this method immediately after `checkCredentialsWithSearchOnlyAPIKey` in several places. Should we just do that automatically from that method and not have to repeat it everywhere? Or make displaying the message an optional boolean parameter in `checkCredentialsWithSearchOnlyAPIKey`?

Reply:
The main advantage of this method is to pass the current class so the user can know where the issue happened. So we would need to pass the information to `checkCredentialsWithSearchOnlyAPIKey` with this change. I think this would unnecessarily complicate the signature of this method with something which has nothing to do with credentials check... 
I also like the fact that the logic within the indexers stays humanly readable ("if my credentials are not speficied, then I display an error message") .

This is called 7 times for now in the codebase, I guess this is something we can deal with, WDTY ?

Reply:
Yeah it's cool. No need to "DRY" ourselves to death. Thanks for thinking about it anyway!